ENH: Style-matching and minor adjustments

thermokarst opened the pull request:
- Removes all `jumbotron`s in favor of `grid`s (to match other Q2 sites)
- Replaces square QIIME 2 logo with rect variant (matching other Q2 sites)
- Clean up iframe border and positioning
- Highlight currently selected tab in navbar
- Match HTML title styling with other Q2 sites
- Update branding position (right) and title (https://view.qiime2.org/)

ebolyen commented on app/ToolbarWindow/Menu.jsx:
```diff
@@ -29,33 +29,37 @@ export default class Menu extends React.Component {
                 <Navbar.Header>
                     <a href='/'>
                         <img style={{height: '40px', marginRight: '10px', marginTop: '4px'}} className='navbar-left' src={logo} alt='QIIME 2'/>
                     </a>
```
Without the brand, the padding gets odd in the menu when you resize to a mobile resolution. (Although the current behavior flips the order of the brand and logo, so it's not doing an amazing job as is)

thermokarst replied:
<img width="356" alt="screen shot 2017-01-03 at 4 54 29 pm" src="https://cloud.githubusercontent.com/assets/274668/21627466/565c153a-d1d5-11e6-9bbf-9b7d8aa2f88f.png">

Just to confirm, is this what you are talking about here? Or is this fun-looking menu something else entirely?

ebolyen replied:
It's on this page, but it isn't the fun-looking menu from the `Navbar.Brand` comment below. If you look closely at the `qiime2` logo, you'll see that it doesn't have the same padding as the rest of the page. This isn't an issue on the current site, so I think moving the `Navbar.Brand` has something to do with it.

thermokarst replied:
Sorry, copied and pasted this comment into the wrong window, meant to ask for the one below, not on the logo padding. Thats what I get for jumping between desktop and mobile.

ebolyen commented on app/ToolbarWindow/Menu.jsx:
```diff
@@ -29,33 +29,37 @@ export default class Menu extends React.Component {
+                        <Navbar.Brand>
+                            {/* Hitting home *should* unload the document */}
+                            <a href='/'>
+                                https://view.qiime2.org/
```
The brand causes the collapse menu to break (in mobile resolution), I don't think it is meant to be inside `Navbar.Collapse`, we could make it a `NavItem` instead though.

thermokarst replied:
:+1:

ebolyen commented on app/ToolbarWindow/Menu.jsx:
```diff
@@ -29,33 +29,37 @@ export default class Menu extends React.Component {
-                    <Navbar.Brand>
-                        {/* Hitting home *should* unload the document */}
-                        <a href='/'>
-                            q2view
```
What if this stayed put and was just `View`? Then it would match the title (the logo + this text would say `QIIME 2` `View`)

thermokarst replied:
View makes sense if we move this back over to the left, which sounds good to me.

ebolyen commented on app/pages/Home/component.jsx:
```diff
@@ -26,27 +26,32 @@ const onDropStyle = {
+                </form>
+            </Col>
+        </Row>
+        <Row><Col md={12} style={ { textAlign: 'center' } }>- or -</Col></Row>
```
The `- or -` is very small now and easy to miss, should we just drop it?

thermokarst replied:
Hmm, renders the same size as before in Chrome, will take a look in FF.

ebolyen commented on app/ToolbarWindow/Content.jsx:
```diff
@@ -5,8 +5,8 @@ import Loader from '../Loader';
-    // "Temporary" hack
-    height: 'calc(100% - 52px)',
+    height: '100%',
+    marginTop: '-20px',
```
This doesn't quite work, I'm still seeing some content clipping in the iframe on the bottom. The "air-quotes" around temporary are there for a reason. Godspeed in your quest!

thermokarst replied:
Bummer, this worked like a charm in Chrome.

thermokarst commented:
Alright, a few updates:

- Replaced the QIIME 2 logo with a new prototype of a "QIIME 2 View" logo (and dropped text label)
- Cleaned up the navbar a bit so that the "hamburger" is hidden when no archive is loaded
- Tweaked the `- or -` text to beef up the size a bit.
- Reverted the `calc` hack back to where it was (with some minor adjustments). FWIW, the only reason that was tweaked in the first place was because I was fiddling with the margins, and one thing led to another, and dropping the `calc` jazz looked "good" (turns out the iframe was just truncated in a convenient location!). Anyway, after a bit of searching, I really can't think of a "better" way to make it happen --- most other solutions seem to do something like this, and this seems to be the least complicated offering!

Thanks @ebolyen! Let me know if you want to see any other changes, or if you uncover any lurking surprises!

<img width="354" alt="screen shot 2017-01-03 at 9 37 08 pm" src="https://cloud.githubusercontent.com/assets/274668/21631815/e3093874-d1fc-11e6-9e1f-dd523b4d7cec.png">
<img width="362" alt="screen shot 2017-01-03 at 9 37 27 pm" src="https://cloud.githubusercontent.com/assets/274668/21631819/eb0a1e12-d1fc-11e6-850d-2b1378ba49b2.png">
<img width="1440" alt="screen shot 2017-01-03 at 9 39 12 pm" src="https://cloud.githubusercontent.com/assets/274668/21631833/1cef50fa-d1fd-11e6-999a-212c7230d287.png">

ebolyen commented on app/ToolbarWindow/Content.jsx:
```diff
@@ -5,9 +5,8 @@ import Loader from '../Loader';
     width: '100%',
-    // "Temporary" hack
-    height: 'calc(100% - 52px)',
-    overflow: 'auto'
```
`overflow: 'auto'` is needed for things like `/incompatible-browser/` in mobile view. Without it we don't get a scrollbar. (You can test this by using private browsing mode in Firefox, which disables service-workers and will bounce you to this page.)

ebolyen replied:
...or you could just navigate to that page it turns out... I've been doing this the hard way!

thermokarst replied:
I had a chrome extension installed that injected a whole boatload of custom css whenever I had devtools up (unbeknownst to me!), and it was applying an `overflow` attr as part of its payload, which is why this looked okay when I dropped it. Good catch!!

ebolyen commented on app/ToolbarWindow/Menu.jsx:
```diff
@@ -4,7 +4,7 @@ import { NavItem, Navbar, Nav, NavDropdown, MenuItem} from 'react-bootstrap';
 import queryString from 'query-string';
 
-import logo from './img/qiime2-square-40.png';
+import logo from './img/q2view.png';
```
This picture is very large, shouldn't it be scaled down a little? Or do retina displays change something here?

thermokarst replied:
This was deliberate --- the displayed image is scaled to `40px` tall, but at a myriad of zoom levels the extra pixels keep things looking nice and crisp (while still looking good shrunk down to `40px` by default). Are you seeing something on your end that looks not so awesome? I'll update the logo to a smaller size (`300px` by `96px`), but not completely shrunk down to `40px`, so that we get a smaller file size (`~115kb` vs `~300kb`).

ebolyen replied:
No it looks fine on modern browsers (though my ancient Opera made it super grainy), I just saw the diff and it seemed a little big. `300x96` sounds great.

ebolyen commented on app/ToolbarWindow/Menu.jsx:
```diff
@@ -29,38 +29,42 @@ export default class Menu extends React.Component {
                     {this.props.hasVisualization && (
-                        <NavItem {...this.makeLinkProps('/visualization', {'type': 'html'})}>
+                        <NavItem {...this.makeLinkProps('/visualization', {'type': 'html'})}
+                         className={(this.props.location.pathname === '/visualization/') ? 'active': ''}>
```
awesome! :+1:

ebolyen commented on app/pages/IncompatibleBrowser/index.js:
```diff
@@ -10,12 +10,10 @@ const downloadFirefox = 'https://www.mozilla.org/firefox'
+        <h1>Incompatible Browser</h1>
+        <p style={{'margin-top': '2em'}}>
+            Sorry, your current browser does not support the latest web-technologies that this site needs.
+        </p>
```
I think there needs to be just a little bit of whitespace on the page after this paragraph.

ebolyen commented:
This is good to go once the last comments are addressed!

ebolyen commented:
Thanks @thermokarst! :tada:

ebolyen commented:
Just pushed to the live site!

ebolyen commented on src/lib/components/NavBar.svelte:
```diff
@@ -33,21 +34,21 @@
+    // Get the height of the navbar to use as an offset for the content. This
+    // height can change if they have compressed the page enough for the navbar
+    // to become a dropdown
+    const nav_bar_height = document.getElementById("navbar")?.clientHeight;
```
https://svelte.dev/docs/svelte/bind#Dimensions

This would make your life easier, and you don't need IDs anymore.

ebolyen commented on src/lib/components/NavBar.svelte:
```diff
@@ -88,9 +89,7 @@
-  {#if !vendored}
-    <NavBanner/>
-  {/if}
+  <NavBanner {vendored}/>
```
Do you still need this change? It seems like it would "just work" with the updated code.

Oddant1 replied:
I don't want the links to be in the NavBanner if it's vendored, so the NavBanner needs to include or not include the links based on the value of vendored

ebolyen replied:
no that makes sense, I mean do you need the `{#if}...{/if}` to be inside `NavBanner` or can you drop the NavBanner entirely (as was done before), since it seems that the reactive height on NavBar is sufficient to account for the 30px that NavBanner makes.

Oddant1 replied:
I'm not sure I understand. We want the empty navbaner to be there on vendored view for stylistic purposes correct? If that's the case then we need the if inside of navbanner to prevent the links from showing if we are vendored.

Are you suggesting we just get rid of the navbanner entirely on vendored view? I don't have strong feelings about this either way.

ebolyen replied:
Yeah, I don't really care about the existence of those 30px, so I would say get rid of it entirely. What I figured the problem was, was the gap underneath the NavBar, which I think is fixed now irrespective of which way the NavBanner ultimately works in vendored mode.

I would say drop it entirely though if its simple.

ebolyen commented on src/lib/components/NavBar.svelte:
```diff
@@ -87,8 +83,8 @@
-<nav id="navbar" use:melt={$root}>
-  {#if !vendored}
+<nav id="navbar" use:melt={$root} bind:offsetHeight={nav_bar_height}>
+  {#if vendored}
```
```suggestion
  {#if !vendored}
```
